src/lang-promql/parser/parser.ts

```diff
@@ -273,6 +264,45 @@ export class Parser {
     }
   }
 
+  private checkVectorSelector(node: Subtree): void {
+    const labelMatchers = buildLabelMatchers(
+      retrieveAllRecursiveNodes(walkThrough(node, LabelMatchers, LabelMatchList), LabelMatchList, LabelMatcher),
+      this.state
+    );
+    // A Vector selector must contain at least one non-empty matcher to prevent
+    // implicit selection of all metrics (e.g. by a typo).
+    let empty = true;
+    let i = 0;
+    while (i < labelMatchers.length && empty) {
+      empty = labelMatchers[i].matchesEmpty();
+      i++;
+    }
```

Comment:
You can just replace the 6 lines above with:

```typescript
const empty = labelMatchers.every(lm => lm.matchesEmpty());
```

...and at that point you can probably just inline it.